### PR TITLE
checklists!

### PR DESCRIPTION
**Pull request by jebeck**
Gonna put some comments in line for you @gniezen 

Can I ping @urmc-costik here too and successfully get his attention??

### CODE REVIEW

File: docs/PumpChecklist.md [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+  - `[ ]` scheduled basal
+    - `[ ]` basal rate intervals with a start time, duration, and rate delivered
+    - `[ ]` name of basal schedule on each scheduled basal rate interval
+  - `[ ]` manual temp basal
+    - `[ ]` basal rate intervals with a start time, duration, and rate delivered
+    - `[ ]` object representing suppressed scheduled basal *for each segment of the basal schedule that the temp basal intersects*
+  - `[ ]` percentage temp basal
+    - `[ ]` basal rate intervals with a start time, duration, percent
+        - `[ ]` rate provided directly OR
+        - `[ ]` rate computed from percent x suppressed.rate
```
Comment (jebeck) [inline docs/PumpChecklist.md]: It doesn't happen very often here, so I'm just using " OR" at the end of a line to indicate when a subset of checklist items are exclusive. If that's not clear enough for you, format-wise, @gniezen LMK if you have another idea for representing suchlike.

File: docs/PumpChecklist.md [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+    - `[ ]` extended bolus that crosses midnight is split into two records
+  - `[ ]` combo/dual bolus
+    - `[ ]` amount of insulin delivered - immediate (normal)
+    - `[ ]` amount of insulin delivered - extended
+    - `[ ]` duration of extended insulin delivery
+    - `[ ]` amount of immediate insulin delivery programmed (if differs from actual delivery, in case of bolus interruption, cancellation, etc.)
+    - `[ ]` amount of extended insulin delivery programmed (if differs from actual delivery, in case of bolus interruption, cancellation, etc.)
+    - `[ ]` duration of extended insulin delivery programmed (if differs from actual duration, in case of bolus interruption, cancellation, etc.)
+    - `[ ]` extended portion of combo bolus that crosses midnight is split into two records
+  - bolus cancellations/interruptions
```
Comment (jebeck) [inline docs/PumpChecklist.md]: I believe the places where I've left off the checkbox next to something (and only included checkboxes in sub-items) make sense. If they don't make sense to you, LMK.

File: docs/PumpChecklist.md [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+    - `[ ]` name of basal schedule on each scheduled basal rate interval
+  - `[ ]` manual temp basal
+    - `[ ]` basal rate intervals with a start time, duration, and rate delivered
+    - `[ ]` object representing suppressed scheduled basal *for each segment of the basal schedule that the temp basal intersects*
+  - `[ ]` percentage temp basal
+    - `[ ]` basal rate intervals with a start time, duration, percent
+        - `[ ]` rate provided directly OR
+        - `[ ]` rate computed from percent x suppressed.rate
+    - `[ ]` object representing suppressed scheduled basal *for each segment of the basal schedule that the temp basal intersects*
+  - `[ ]` "suspended" basals (see [status - suspends & resumes)[#device-events] below)
```
Comment (gniezen) [inline docs/PumpChecklist.md]: typo in link?

File: docs/PumpChecklist.md (hunk 1 above)
Comment (gniezen) [inline docs/PumpChecklist.md]: I'm happy with this format-wise.

File: docs/PumpChecklist.md [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+    - `[ ]` name of basal schedule OR
+    - `[ ]` name of settings profile
+    - `[ ]` each schedule as a set of objects each with a rate and a start time
+  - `[ ]` name of currently active basal schedule
+  - `[ ]` units of all blood glucose-related fields (read from device, not hard-coded)
+  - `[ ]` units of all carb-related fields (read from device, not hard-coded)
+  - `[ ]` carb ratio(s)
+    - `[ ]` name of settings profile
+    - `[ ]` (one or more) set(s) of objects each with a ratio (amount) and a start time
+  - `[ ]` insulin sensitivity factor(s)
```
Comment (gniezen) [inline docs/PumpChecklist.md]: What if there's no name for the settings profile, as there's only one set of settings?

File: docs/PumpChecklist.md (hunk 3 above)
Comment (jebeck) [inline docs/PumpChecklist.md]: Good catch - fixing on next update.

File: docs/PumpChecklist.md (hunk 4 above)
Comment (jebeck) [inline docs/PumpChecklist.md]: All of the things here are possibly not implemented by the data protocol/spec. Just leave the checkbox item blank (as per key at top) if not part of a device's data protocol/spec.

**Comment by jebeck** [comment]
There are completed or draft filled-in checklists for all pumps currently supported or in development. Have not yet drafted checklists for CGM data through Animas or Tandem, but those can wait a bit I think. Merging.

**Pull request merged**